codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/endpoint/EndpointsDecorator.kt

```diff
@@ -79,6 +79,14 @@ interface EndpointCustomization {
      * Provide a list of additional endpoints standard library functions that rules can use
      */
     fun customRuntimeFunctions(codegenContext: ClientCodegenContext): List<CustomRuntimeFunction> = listOf()
+
+    /**
+     * Allows overriding the `finalize_params` method  in the `ResolveEndpoint` trait.
+     *
+     * `ResolveEndpoint::finalize_params` provides a default no-op implementation,
+     * and this customization enables the implementor to provide an alternative implementation.
+     */
+    fun overrideFinalizeEndpointParams(codegenContext: ClientCodegenContext): Writable? = null
```

Comment:
What happens if multiple customizations implement this?

Reply:
Made customizations append-able and updated the doc in 280294c.